File: src/main/java/com/ricardococati/calculacotacoes/utils/converters/SplitInplitConverter.java
package com.ricardococati.calculacotacoes.utils.converters;

import com.ricardococati.calculacotacoes.entities.domains.split.SplitInplit;
import com.ricardococati.calculacotacoes.entities.enums.OperacaoSplitInplit;
import java.time.LocalDate;
import org.springframework.stereotype.Component;

@Component
public class SplitInplitConverter {

  public SplitInplit convert(
      final LocalDate dtPregao,
      final String codneg,
      final Integer qtdSplitInplit,
      final String operacao) {
    return SplitInplit
        .builder()
        .codneg(codneg)
        .dtpreg(dtPregao)
        .qtdSplitInplit(qtdSplitInplit)
        .operacao(OperacaoSplitInplit.valueOf(operacao.toUpperCase()))
        .build();
  }

}

File: application/src/main/kotlin/com/ricardococati/controller/converter/SplitInplitConverter.kt
package com.ricardococati.controller.converter

import com.ricardococati.model.entities.SplitInplit
import com.ricardococati.model.enums.OperacaoSplitInplit
import org.springframework.stereotype.Component
import java.time.LocalDate

@Component
class SplitInplitConverter {

    fun convert(
            dtPregao: LocalDate?,
            codneg: String?,
            qtdSplitInplit: Int?,
            operacao: String): SplitInplit {
        return SplitInplit
                .builder()
                .codneg(codneg)
                .dtpreg(dtPregao)
                .qtdSplitInplit(qtdSplitInplit)
                .operacao(OperacaoSplitInplit.valueOf(operacao.toUpperCase()))
                .build()
    }

}